hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestUnsetAndChangeDirectoryEcPolicy.java

```diff
@@ -388,4 +390,31 @@ public void testUnsetEcPolicyInEditLog() throws IOException {
     cluster.restartNameNode(true);
     Assert.assertNull(fs.getErasureCodingPolicy(new Path("/")));
   }
+
+  /*
+   * Test Client unset EC policy on directory.
+   */
+  @Test
+  public void testClientUnsetEcPolicy() throws Exception {
+    ClientProtocol clientProtocol = fs.getClient().getNamenode();
+    clientProtocol.setErasureCodingPolicy("/",ErasureCodeConstants.REPLICATION_POLICY_NAME);
+    final String dirString = "/striped";
+    clientProtocol.mkdirs(dirString, null, false);
```

Comment:
It gave me a NPE here. 
I changed it to 
``
fs.mkdirs(new Path(dirString));
``
to make it work.

But this test passes without your fix as well. So, this isn't reproducing the issue

Reply:
Thanks Reply. The issue is fake, my fault. I will close the PR.